### ISSUE TITLE
[Safe Autocorrect Bug] Style/RedundantParentheses swallows chained methods into inline comments

### ISSUE THREAD

**Issue by thai-eh** [rubocop/rubocop#15135]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Style/RedundantParentheses

### Steps to reproduce the problem

When Style/RedundantParentheses autocorrects a multiline expression, it deletes the closing parenthesis and the preceding newline character. If the inner expression ends with an inline comment, this text-deletion causes any trailing chained methods to be pulled up into the comment, breaking the code or swallowing linter directives.

Steps to Reproduce
1. Create a file named example.rb with the following code:
```ruby
member_ids = (
  [1, 2, 3].map { |i| i } # rubocop:disable Rails/Pluck
).uniq
```
2. Run RuboCop with autocorrect for this specific cop:
```sh
bundle exec rubocop example.rb --only Style/RedundantParentheses -A
```

### Expected behavior

The autocorrect should either preserve the execution of .uniq by handling the comment safely, or it should safely bail out and refuse to autocorrect if an inline comment is detected at the end of the begin node's line.

Example of safe correction:
```ruby
member_ids = [1, 2, 3].map { |i| i } # rubocop:disable Rails/Pluck
  .uniq
```

### Actual behavior

The newline and closing parenthesis are blindly deleted, causing .uniq to become part of the inline comment. The method is no longer executed, and the RuboCop disable directive becomes invalid.
```ruby
member_ids = [1, 2, 3].map { |i| i } # rubocop:disable Rails/Pluck.uniq
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.84.2
```

**Comment by hammadxcm** [comment]
I've opened #15140 with a fix for this. The corrector now preserves the newline before `)` when the line above ends with an inline comment and chained calls follow `)`, so the chain stays valid and the comment (including any rubocop directive) remains intact.
